src/osd/OSD.cc

```diff
@@ -5536,6 +5541,27 @@ void OSD::_send_boot()
   set_state(STATE_BOOTING);
 }
 
+void OSD::_collect_compression_algorithms(map<string, string>& pm)
+{
+  const auto& compression_algorithm_names = Compressor::compression_algorithm_names;
+  const auto& plugin_registry = cct->get_plugin_registry()->plugins;
+
+  ostringstream os;
+  for_each(begin(compression_algorithm_names), end(compression_algorithm_names),
+  		  [&os, &plugin_registry](const auto& algorithm_name) {
+  		   if (plugin_registry.end() != plugin_registry.find(algorithm_name))
+  			os << algorithm_name << ',';
+  		  });
+  
+  string s(os.str());
+  
+  // No ostream_joiner yet, so we'll need to do some trimming:
+  if(s[s.size()] == ',')
```

Comment:
This should probably be ```size() - 1```?

Reply:
Good catch! "Someone" didn't have enough coffee! Thanks.